test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedArchiveTestUtility.cs

```diff
@@ -146,5 +148,153 @@ internal static class SignedArchiveTestUtility
+            ReadAndWriteUntilPosition(reader, writer, metadata.StartOfFileHeaders);
+
+            // Write all file entries in the new order
+            foreach (var entry in shiftedCdr)
+            {
+                reader.BaseStream.Seek(offset: entry.OffsetToFileHeader, origin: SeekOrigin.Begin);
+                ReadAndWriteUntilPosition(reader, writer, entry.OffsetToFileHeader + entry.FileEntryTotalSize);
+            }
+
+            // Write all central directory records with updated offsets
+            shiftedCdr.Sort((x, y) => Comparer<long>.Default.Compare(x.IndexInRecords, y.IndexInRecords));
+
+            reader.BaseStream.Seek(offset: metadata.StartOfCentralDirectory, origin: SeekOrigin.Begin);
+            foreach (var entry in shiftedCdr)
+            {
+                reader.BaseStream.Seek(offset: entry.Position, origin: SeekOrigin.Begin);
+                ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Position + 28);
+
+                var filenameLength = reader.ReadUInt16();
+                writer.Write(filenameLength);
+
+                var extraFieldLength = reader.ReadUInt16();
+                writer.Write(extraFieldLength);
+
+                var fileCommentLength = reader.ReadUInt16();
+                writer.Write(fileCommentLength);
+
+                ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Position + 8);
+
+                var relativeOffsetOfLocalFileHeader = (uint)(reader.ReadUInt32() + entry.ChangeInOffset);
+                writer.Write(relativeOffsetOfLocalFileHeader);
+
+                ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Position + filenameLength + extraFieldLength + fileCommentLength);
+            }
+
+            // Write everything after central directory records
+            reader.BaseStream.Seek(offset: metadata.EndOfCentralDirectory, origin: SeekOrigin.Begin);
+            ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Length);
+
+            return Task.FromResult(0);
+        }
+
+        private static List<CentralDirectoryMetadata> ShiftMetadata(
+            SigningSpecifications spec,
+            SignedPackageArchiveMetadata metadata,
+            int newSignatureFEIndex,
+            int newSignatureCDRIndex)
+        {
+            var shiftedCdr = new List<CentralDirectoryMetadata>(metadata.CentralDirectoryRecords);
+
+            shiftedCdr.Sort((x, y) => Comparer<long>.Default.Compare(x.OffsetToFileHeader, y.OffsetToFileHeader));
+            ShiftSignatureCDRToIndex(spec, shiftedCdr, newSignatureFEIndex);
+
+            var lastEntryEnd = 0L;
+            foreach (var cdr in shiftedCdr)
+            {
+                cdr.ChangeInOffset = lastEntryEnd - cdr.OffsetToFileHeader;
+
+                lastEntryEnd = cdr.OffsetToFileHeader + cdr.FileEntryTotalSize + cdr.ChangeInOffset;
+            }
+
+            shiftedCdr.Sort((x, y) => Comparer<long>.Default.Compare(x.Position, y.Position));
```

Comment:
This can be simplified - 
`shiftedCdr.Sort((x, y) => x.Position.CompareTo(y.Position));`